src/actions/posts.js

```diff
@@ -1095,6 +1095,29 @@ export function doPostAction(postId, actionId, selectedOption = '') {
                 data: data.trigger_id,
             });
         }
+
+        return {data};
+    };
+}
+
+export function doPostActionWithCookie(postId, actionId, actionCookie, selectedOption = '') {
```

Comment:
Maybe we can rewrite `doPostAction` to use `doPostActionWithCookie(postId, actionId, null, selectedOption)` or something like that.